networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/Eth2PeerManager.java

```diff
@@ -66,6 +70,12 @@ public void onConnect(@NotNull final Peer peer) {
     eth2Peer.subscribeInitialStatus(
         (status) ->
             PeerChainValidator.create(storageClient, historicalChainData, eth2Peer, status).run());
+
+    connectSubscribers.forEach(c -> c.onConnected(eth2Peer));
```

Comment:
Do we want to consider the peer connected before we get the initial status? I'm somewhat inclined to treat peers as completely non-existent until we've done the basic sanity checks (ideally including checking they're on the same chain).

Reply:
Good catch - we definitely want to wait until the peer is validated before firing these events.  Will fix.